File: src/Java/CreationalPatterns/Builder/Laptops/src/ConcreteBuilders/LenovoConcreteBuilder.java
package ConcreteBuilders;

import Builder.LaptopBuilder;
import Product.Laptop;

import java.util.Scanner;

public class LenovoConcreteBuilder extends LaptopBuilder {
    private Scanner scanner;

    public LenovoConcreteBuilder() {
        laptop = new Laptop();
        scanner = new Scanner(System.in);
    }

    @Override
    public void setModel() {
        System.out.print("Enter Laptop Model: ");
        laptop.setModel(scanner.nextLine());
    }

    @Override
    public void setCPUSeries() {
        System.out.print("Enter CPU Series: ");
        laptop.setCPUSeries(scanner.nextLine());
    }

    @Override
    public void setGPUModel() {
        System.out.print("Enter GPU Model: ");
        laptop.setGPUModel(scanner.nextLine());
    }

    @Override
    public void setRAMType() {
        System.out.print("Enter RAM Type: ");
        laptop.setRAMType(scanner.nextLine());
    }

    @Override
    public void setRAMSize() {
        System.out.print("Enter RAM Size: ");
        laptop.setRAMSize(Integer.parseInt(scanner.nextLine()));
    }

    @Override
    public void setDisplayType() {
        System.out.print("Enter Display Type: ");
        laptop.setDisplayType(scanner.nextLine());
    }

    @Override
    public void setSSDType() {
        System.out.print("Enter SSD Type: ");
        laptop.setSSDType(scanner.nextLine());
    }

    @Override
    public void setSSDCapacity() {
        System.out.print("Enter SSD Capacity: ");
        laptop.setSSDCapacity(Integer.parseInt(scanner.nextLine()));
    }

    @Override
    public void setExtras() {
        System.out.print("Add extra item (Exit with END): ");
        String extraItem = scanner.nextLine();

        while (!extraItem.equals("END")) {
            laptop.getExtras().add(extraItem);
            System.out.print("Add extra item (Exit with END): ");
            extraItem = scanner.nextLine();
        }
    }
}

File: src/PHP/CreationalPatterns/Builder/Laptops/ConcreteBuilders/LenovoConcreteBuilder.php
<?php

namespace CreationalPatterns\Builder\Laptops\ConcreteBuilders;

use CreationalPatterns\Builder\Laptops\Builder\LaptopBuilder;
use CreationalPatterns\Builder\Laptops\Product\Laptop;

class LenovoConcreteBuilder extends LaptopBuilder
{
    public function __construct()
    {
        $this->laptop = new Laptop();
    }

    public function setModel(): void
    {
        echo "Enter Laptop Model: ";
        $this->laptop->model = readline();
    }

    public function setCPUSeries(): void
    {
        echo "Enter CPU Series: ";
        $this->laptop->cpuSeries = readline();
    }

    public function setGPUModel(): void
    {
        echo "Enter GPU Model: ";
        $this->laptop->gpuModel = readline();
    }

    public function setRAMType(): void
    {
        echo "Enter RAM Type: ";
        $this->laptop->ramType = readline();
    }

    public function setRAMSize(): void
    {
        echo "Enter RAM Size: ";
        $this->laptop->ramSize = floatval(readline());
    }

    public function setDisplayType(): void
    {
        echo "Enter Display Type: ";
        $this->laptop->displayType = readline();
    }

    public function setSSDType(): void
    {
        echo "Enter SSD Type: ";
        $this->laptop->ssdType = readline();
    }

    public function setSSDCapacity(): void
    {
        echo "Enter SSD Capacity: ";
        $this->laptop->ssdCapacity = floatval(readline());
    }

    public function setExtras(): void
    {
        echo "Add extra item (Exit with END): ";
        $extraItem = readline();

        while ($extraItem !== "END")
        {
            $this->laptop->extras[] = $extraItem;
            echo "Add extra item (Exit with END): ";
            $extraItem = readline();
        }
    }
}